storage/innobase/xtrabackup/test/t/check_for_passwords.sh

```diff
@@ -15,22 +18,31 @@ mysql -e "CREATE TABLE t1(i int) engine=MYISAM" test
 mysql -e "INSERT INTO t1 values(1)" test
 mkdir -p $topdir/backup
 
-#pass option like password -p to be run with 
+#pass option like password -p to be run with
+
 function grep_in_datadir_ps() {
+
   additional_options=$1
-  mysql -e "SELECT SLEEP(15) from t1 for update" test &
+  mysql -e "SELECT SLEEP(1000) from t1 for update " test &
+
   $XB_BIN $XB_ARGS --backup -u pxb $additional_options\
   --transition-key=$password_string --target-dir=$topdir/backup \
   2>&1 | tee $topdir/pxb.log &
   job_pid=$!
-  #sleep sufficient time to let backup run
-  sleep 5
-  vlog "check for $password_string string in ps "
-  if ps -ef | grep "xtrabackup.*$password_string" | grep -v grep
+
+  wait_for_process_to_start $job_pid
+  #sleep is to ensure password is removed by xtrabackup
+  sleep 3
```

Comment:
Does this mean there is window that xtrabackup shows password?

Reply:
yes there is window when xtrabackup shows password.  PXB removes the argument passed to script.

Reply:
afair, this is done as soon as soon the argument processing is done. Shouldn't it be less than fraction of second? does this test fail without the sleep?

Reply:
yes this is done after argument processing. yes test fails randomly without sleep.

Reply:
can you please check (by code or test) if the same behaviour applies for "mysql" client tool as well?

Reply:
yes. same thing happen with mysql client as well.

Reply:
ok